### PR TITLE
[FEATURE] TYPO3 12 compatibility

### PR DESCRIPTION
**Pull request by alexanderschnitzler**
If this PR were to merged, the extension would be compatible with TYPO3 12 but it also removes compatibility with lower versions because:

- dependency to `helhum/typo3-console` has been removed
- command is registered via Di instead of `Commands.php`
- min versions of other requirements have been raised according to what TYPO3 12 supports

This PR also sets version to 2 in `ext_emconf.php`, which would be in line with a new 2.0.0 tag on packagist.

Great extension btw.! <3

### CODE REVIEW

**Review by bmack — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
 	"name": "bmack/site-importer",
 	"description": "Imports records from a Yaml file structure into the TYPO3 database",
 	"license": "GPL-2.0+",
-	"type": "typo3-console-command",
+	"type": "typo3-cms-extension",
```
Comment (bmack) [inline composer.json:5]: Changing this would mean, I should then register the extension key ;) which I just did.

**Comment by alexanderschnitzler** [comment]
Hi Benni,
since you approved the changes, I wonder what's the next step with this MR. Do you plan to merge and release a new version or do you need any changes from my side?

Have a nice day. :)

**Comment by bmack** [comment]
sorry, was too distracted with other stuff the last few days, and forgot about this one.

**Comment by alexanderschnitzler** [comment]
No need to be sorry. Thanks for merging!

**Pull request merged**